ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -57,30 +57,37 @@
   private final long maxNumberOfFiles;
   private final boolean hiveInTest;
   private final String copyAsUser;
+  private FileSystem destinationFs;
 
-  public CopyUtils(String distCpDoAsUser, HiveConf hiveConf) {
+  public CopyUtils(String distCpDoAsUser, HiveConf hiveConf, FileSystem destinationFs) {
     this.hiveConf = hiveConf;
     maxNumberOfFiles = hiveConf.getLongVar(HiveConf.ConfVars.HIVE_EXEC_COPYFILE_MAXNUMFILES);
     maxCopyFileSize = hiveConf.getLongVar(HiveConf.ConfVars.HIVE_EXEC_COPYFILE_MAXSIZE);
     hiveInTest = hiveConf.getBoolVar(HiveConf.ConfVars.HIVE_IN_TEST);
     this.copyAsUser = distCpDoAsUser;
+    this.destinationFs = destinationFs;
   }
 
   // Used by replication, copy files from source to destination. It is possible source file is
   // changed/removed during copy, so double check the checksum after copy,
   // if not match, copy again from cm
-  public void copyAndVerify(FileSystem destinationFs, Path destRoot,
+  public void copyAndVerify(Path destRoot,
                     List<ReplChangeManager.FileInfo> srcFiles) throws IOException, LoginException, HiveFatalException {
     Map<FileSystem, Map< Path, List<ReplChangeManager.FileInfo>>> map = fsToFileMap(srcFiles, destRoot);
     UserGroupInformation proxyUser = getProxyUser();
     try {
       for (Map.Entry<FileSystem, Map<Path, List<ReplChangeManager.FileInfo>>> entry : map.entrySet()) {
-        FileSystem sourceFs = entry.getKey();
         Map<Path, List<ReplChangeManager.FileInfo>> destMap = entry.getValue();
         for (Map.Entry<Path, List<ReplChangeManager.FileInfo>> destMapEntry : destMap.entrySet()) {
           Path destination = destMapEntry.getKey();
           List<ReplChangeManager.FileInfo> fileInfoList = destMapEntry.getValue();
-          boolean useRegularCopy = regularCopy(destinationFs, sourceFs, fileInfoList);
+          if (fileInfoList.isEmpty()) {
```

Comment:
In what scenario, fileInfoList can be empty?

Reply:
currently no ..but if some filters are added and the files to be copied are removed then this could lead to empty list

Reply:
In that case, can we just change it to assert? and add a comment for this assumption.

Reply:
i think instead of assert putting a log is better option as the scenario is handled

Reply:
I think, handling this case is redundant if it doesn't exist today and it is dead code. Also, assert is the best way to represent such assumptions. I think, this is just a copy utility class which is expected to copy the list of source files to target directory. If at all any filtering to be done, it should be done before we make this map.

Reply:
done